src/vulkan-renderer/wrapper/device.cpp

```diff
@@ -34,6 +32,95 @@ constexpr float default_queue_priority = 1.0f;
 
 namespace inexor::vulkan_renderer::wrapper {
 
+bool Device::device_extension_supported(const VkPhysicalDevice graphics_card, const std::string &extension_name) {
+    assert(graphics_card);
+    assert(!extension_name.empty());
+
+    std::uint32_t device_extension_count = 0;
+
+    // Query how many device extensions are available.
+    if (const auto result =
+            vkEnumerateDeviceExtensionProperties(graphics_card, nullptr, &device_extension_count, nullptr);
+        result != VK_SUCCESS) {
+        throw exceptions::VulkanException("Error: vkEnumerateDeviceExtensionProperties failed!", result);
+    }
+
+    std::vector<VkExtensionProperties> device_extensions(device_extension_count);
+
+    if (device_extension_count == 0) {
+        throw std::runtime_error("Error: No Vulkan device extensions available!");
+    } else {
```

Comment:
No need for `else` after `throw`.

Reply:
I will apply this to every code block in instance and device wrapper.

Reply:
Done.